File: src/main/java/custom/tibame201020/adbOpenCv/script/robot/RobotScript.java
package custom.tibame201020.adbOpenCv.script.robot;

import custom.tibame201020.adbOpenCv.robot.RobotService;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Component;

@Component
@Lazy
public class RobotScript {

    private final RobotService robotService;

    public RobotScript(RobotService robotService) {
        this.robotService = robotService;
    }

    public void execute() throws Exception {
        System.setProperty("java.awt.headless", "false");

        var testSnapshot = "robot-snapshot.png";
        var snapshotPath = robotService.snapshot(testSnapshot);
        System.err.println(snapshotPath);

        var result = robotService.findImage(testSnapshot);
        System.err.printf("x: %s, y: %s", result.point().x, result.point().x);
    }
}

File: core/script/robot/robot_script.py
"""
Robot 測試腳本
"""
from service.platform.robot.robot_platform import RobotPlatform


class RobotScript:
    """Robot 測試腳本"""
    
    def __init__(self, robot_platform: RobotPlatform):
        self.robot_platform = robot_platform
    
    def execute(self) -> None:
        """執行腳本"""
        test_snapshot = "robot-snapshot.png"
        snapshot_path = self.robot_platform.snapshot(test_snapshot)
        print(snapshot_path)
        
        result = self.robot_platform.find_image(test_snapshot)
        print(f"x: {result.point[0]}, y: {result.point[1]}")

